src/mesh/data/gridfromfile.cxx

```diff
@@ -194,6 +194,11 @@ bool GridFile::getField(Mesh* m, T& var, const std::string& name, BoutReal def,
     break;
   }
   case 3: {
+    if (not option.is_loaded) {
+      size[0] = option.shape[0];
```

Comment:
> Two things:
> 
>     1. can we just change `size` to be `std::vector<std::size_t>` to avoid this warning?

I tried this. We need to change `int` to `size_t` in all the `getField` functions. This then results in a lot of sign-compare warnings. I do not have the motivation to continue changing a lot of code ...

>     2. can we pull out this whole bit into a helper function to keep `size` `const` in the rest of this function? Then we could just `return option.shape`?

Yes, I moved the whole size business to `Options`